gazelle/manifest/manifest.go

```diff
@@ -40,12 +40,17 @@ func NewFile(manifest *Manifest) *File {
 }
 
 // Encode encodes the manifest file to the given writer.
-func (f *File) Encode(w io.Writer, manifestGeneratorHashFile, requirements io.Reader) error {
+func (f *File) EncodeWithIntegrity(w io.Writer, manifestGeneratorHashFile, requirements io.Reader) error {
 	integrityBytes, err := f.calculateIntegrity(manifestGeneratorHashFile, requirements)
 	if err != nil {
 		return fmt.Errorf("failed to encode manifest file: %w", err)
 	}
 	f.Integrity = fmt.Sprintf("%x", integrityBytes)
+
+	return f.EncodeWithoutIntegrity(w)
+}
+
+func (f *File) EncodeWithoutIntegrity(w io.Writer) error {
```

Comment:
[nit] This function should be `encode`, then you can have both `EncodeWithIntegrity` and `EncodeWithoutIntegrity` as exported functions that call the internal `encode`. It just reads better than `EncodeWithIntegrity` calling `EncodeWithoutIntegrity`.

Reply:
Updated!